src/machines/modelingMachine.ts

```diff
@@ -2686,6 +2688,33 @@ export const modelingMachine = setup({
         )
       }
     ),
+    importAstMod: fromPromise(
+      async ({ input }: { input?: ModelingCommandSchema['Import'] }) => {
+        if (!input) {
+          return new Error('No input provided')
+        }
+
+        const ast = kclManager.ast
+        const { path, localName } = input
+        const { modifiedAst, pathToNode } = addImport({
+          node: ast,
+          path,
+          localName,
+        })
+        await updateModelingState(
+          modifiedAst,
+          EXECUTION_TYPE_REAL,
+          {
+            kclManager,
+            editorManager,
+            codeManager,
+          },
+          {
+            focusPath: [pathToNode],
+          }
+        )
```

Comment:
For some reason this isn't working properly yet, I've tracked it down to step 2 > updateEditorWithAstAndWriteToFile > updateCodeStateEditor > updateCodeEditor > editorManager.editorView.dispatch, which is the CM call and it appears to be failing but can't explain why yet.

Reply:
Solved it through a flag in the updateModelingScene function, left a TODO.